Mathlib/Probability/Moments/Tilted.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2025 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import Mathlib.MeasureTheory.Measure.Tilted
+import Mathlib.Probability.Moments.MGFAnalytic
+
+/-!
+# Results relating `Measure.tilted` to `mgf` and `cgf`
+
+For a random variable `X : Ω → ℝ` and a measure `μ`, the tilted measure `μ.tilted (t * X ·)` is
+linked to the moment generating function (`mgf`) and the cumulant generating function (`cgf`)
+of `X`.
+
+## Main statements
+
+* `integral_tilted_mul_self`: the integral of `X` against the tilted measure `μ.tilted (t * X ·)`
+  is the first derivative of the cumulant generating function of `X` at `t`.
+  `(μ.tilted (t * X ·))[X] = deriv (cgf X μ) t`
+* `variance_tilted_mul`: the variance of `X` under the tilted measure `μ.tilted (t * X ·)`
+  is the second derivative of the cumulant generating function of `X` at `t`.
+  `Var[X; μ.tilted (t * X ·)] = iteratedDeriv 2 (cgf X μ) t`
+
+-/
+
+
+open MeasureTheory Real Set Finset
+
+open scoped NNReal ENNReal ProbabilityTheory
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {μ ν : Measure Ω} {X : Ω → ℝ} {t u : ℝ}
+
+namespace ProbabilityTheory
+
+section Apply
+
+/-! ### Apply lemmas for `tilted` expressed with `mgf` or `cgf`. -/
+
+lemma tilted_mul_apply_mgf' {s : Set Ω} (hs : MeasurableSet s) :
+    μ.tilted (t * X ·) s = ∫⁻ a in s, ENNReal.ofReal (exp (t * X a) / mgf X μ t) ∂μ := by
+  rw [tilted_apply' _ _ hs, mgf]
+
+lemma tilted_mul_apply_mgf [SFinite μ] (s : Set Ω) :
+    μ.tilted (t * X ·) s = ∫⁻ a in s, ENNReal.ofReal (exp (t * X a) / mgf X μ t) ∂μ := by
+  rw [tilted_apply, mgf]
+
+lemma tilted_mul_apply_cgf' [NeZero μ] {s : Set Ω} (hs : MeasurableSet s)
```

Comment:
You shouldn't need `NeZero` here, because the result is obviously true for the zero measure. Something like
```
  rcases eq_zero_or_neZero μ with rfl | hμ 
  · simp
  · simp_rw [tilted_mul_apply_mgf' hs, exp_sub, exp_cgf ht]
```
Same thing below.